radixdlt/src/integration/java/com/radixdlt/consensus/PerfectNetworkTest.java

```diff
@@ -83,4 +81,37 @@ public void given_4_correct_bfts__then_all_should_get_same_commits_consecutive_v
 			.take(time, timeUnit)
 			.blockingSubscribe();
 	}
+
+	@Test
+	public void given_a_correct_bft__then_new_memory_used_should_be_less_than_10MB_over_2_minutes() {
+		final int numNodes = 1;
+		final long time = 2;
+		final TimeUnit timeUnit = TimeUnit.MINUTES;
+
+		final List<ECKeyPair> nodes = createNodes(numNodes);
+		final BFTTestNetwork bftNetwork = new BFTTestNetwork(nodes);
+
+		Observable<Long> memory = Observable.interval(500, TimeUnit.MILLISECONDS)
+			.map(l -> {
+				Runtime.getRuntime().gc();
```

Comment:
From what I remember, `Runtime.gc()` is not guaranteed to do anything before it returns (or anything at all, for that matter, since it's just a hint to the GC). So this may not work the way it was intended to.